[FEAT] 매직 넘버 심사를 위한 회원가입 로직에 관련 로직 추가

JungYoonShin opened the pull request:
<!--
name: Makers Auth Feature PR template
about: 구현한 기능과 변경 사항에 대해 구체적으로 작성해주세요
title: '[FEAT/{#이슈번호}] 기능 내용'
labels: ''
assignees: ''
-->

<!--
- 리뷰어 추가했나요?
- 허가자 추가했나요?
- 라벨 추가했나요?
-->

## Related Issue 🚀
- closed #132 

## Work Description ✏️
- 매직 넘버 심사를 위한 회원가입 로직에 관련 로직 추가
  - 최종 Flow는 아래와 같아요! (`signUpForMagicNumber`와 `signUpForSoptUser` 메서드 분리)
     - 매직넘버 유저는 Users DB에 이미 존재하고 있는 상태 (테스트용 유저의 계정은 하나로 유지)
     - 기존 테스트 계정 조회 후 소셜 플랫폼 및 auth_platform_id 업데이트


## PR Point 📸
<!-- 피드백을 받고 싶은 부분을 적어주세요. -->

JungYoonShin committed:
[FEAT] 매직 넘버 심사를 위한 회원가입 로직에 관련 로직 추가

sung-silver commented on src/main/java/sopt/makers/authentication/application/service/auth/SignUpService.java:
```diff
@@ -49,6 +53,9 @@ public void signUp(SignUpCommand command) {
 
     userActivityHistoryRepository.save(savedUser, activity);
+    if (isMagicPhone(command.phone())) {
+      return;
```
`p2`

제 생각에는 저희가 테스트용 유저 계정은 한개로 유지하기로 했었기에, 새로운 유저를 생성하는 것이 아니라 magic 넘버일 경우에는 기존에 테스트용 유저의 계정을 조회하고 그 계정의 소셜 플랫폼 및 id를 업데이트하는 로직이 있어야할 것 같은데 어떻게 생각하시나요?

그래서 signUp 로직에서 매직넘버인지 아닌지를 처음에 구분하여 private 메서드로 각각의 상황에 맞게 로직이 진행되도록 분리하는게 어떨지 제안드립니다

JungYoonShin replied:
놓친 부분이었는데, 짚어주셔서 감사해요🚀 반영했습니당!

JungYoonShin committed:
[FIX] 매직넘버 회원가입 로직 수정

- 매직넘버 회원가입의 경우와 아닌 경우를 분리합니다.
- 매직넘버 회원가입의 경우 기존 회원정보의 social account를 update합니다.

sung-silver commented on src/main/java/sopt/makers/authentication/application/service/auth/SignUpService.java:
```diff
@@ -30,19 +30,34 @@ public class SignUpService implements SignUpUsecase {
+
+  private void signUpForSoptUser(String phone, AuthPlatform authPlatform) {
+    //    String authPlatformId =
+    //            oAuthAuthenticator.getIdentifier(command.token(), command.authPlatform());
```
`p1`
이부분 주석 해제해야하지 않나요?

sung-silver commented on src/main/java/sopt/makers/authentication/application/service/auth/SignUpService.java:
```diff
@@ -30,19 +30,34 @@ public class SignUpService implements SignUpUsecase {
+            .findByPhone(phone)
             .orElseThrow(() -> new AuthException(NOT_FOUND_REGISTER_INFO));
-    SocialAccount socialAccount = createSocialAccount(authPlatformId, command.authPlatform());
+    SocialAccount socialAccount = createSocialAccount(auth, authPlatform);
```
`p2`
위의 주석 해제하고 여기서 authPlatformId를 반환해야 올바른 SocialAccount 도메인 생성이 가능할 것 같습니다!

sung-silver commented on src/main/java/sopt/makers/authentication/application/service/auth/SignUpService.java:
```diff
@@ -70,4 +85,8 @@ private Profile createProfile(UserRegisterInfo registerInfo) {
     return Activity.of(registerInfo.getGeneration(), null, registerInfo.getPart());
   }
+
+  private boolean isMagicPhone(String phone) {
+    return (phone.equals(magicLoginProperty.phone()));
+  }
```
`p3`
연진님이 이전에 전화번호 인증에서 매직넘버인지 확인하는 과정에 해당 메서드와 동일한 메서드를 선언해서 사용하셨는데 클래스를 따로 빼서 public 메서드로 만드는 것은 어떨까요??

JungYoonShin replied:
그렇게 하면 재사용성에는 좋을 것 같은데, 매직넘버의 경우에는 확장될 여지가 없을 것 같아서 요 하나의 메서드를 위해서 클래스를 따로 빼는 것보다 유지해도 될 것 같다고 생각이 들긴 합니다..!

연진이의 생각은 어떤가용?? @seoyeonjin

seoyeonjin replied:
현재는 반복되는 부분이 한 곳뿐이고, 확장 가능성이 낮아 보여서 지금 방식으로 유지해도 괜찮을 것 같습니다.
다만 이후에 비슷한 로직이 반복적으로 생긴다면 Util 클래스로 분리해 관리하는 것도 좋은 방법일 것 같아요!

JungYoonShin committed:
[FIX] 주석으로 인한 에러 문제 해결
